### DISCUSSION TITLE
What is the use of EJBCA?

### DISCUSSION THREAD

**Discussion by agarm9** [Keyfactor/ejbca-ce#601] [Q&A]
I want to learn more about EJBCA. I know it is open source PKI but what is the advantage of using it? How it is different than Microsoft PKI. On the pricing point it charges 2.5$ per hour apart from the OS running cost. So, Microsoft PKI comes free with OS, then why would someone pay extra?

My question may be silly but I am very new to EJBCA and hence seeking guidance.

**Comment by primetomas** [comment]
First, EJBCA Community (open source) you run for free. You're looking at EJBCA Enterprise Cloud I guess when you mention a price?
Windows OS isn't free, and not everyone uses Windows. Even if, there are lots of technical reasons why ADCS is not suitable for many organizations. You will find lots of information if you search the internet.
Here are some things that Keyfactor write, but other vendors will have similar messages.
https://www.keyfactor.com/microsoft-pki/

**Reply by agarm9** [reply]
Thanks for your response.

Yes, I was talking about EJBCA enterprise. I understand windows OS is not free, but what I meant is that once I purchase the OS then I can install PKI for free. It won't make sense for anyone to run EJBCA on windows device. 

Let me reframe my question. Is there any customers who are using EJBCA on a windows machine? If yes, what benefit they get which they can't get in ADCS.

**Reply by primetomas** [reply]
Yes there are many that uses EJBCA on Windows. For the reasons highlighted in that article. For example: with ADCS you need one server per CA, and the average organization have 8 CAs, with some having >30. That's a lot of windows servers to manage when you don't have multi-CA capabilities.
To be fair of course, most organizations are moving to more modern architectures nowadays with containers and cloud deployment where Windows servers are less used. En EJBCA running in Azure for example can serve PKI for all your on-premise windows domains, including Intune. Many organization have pretty complex environments today and really need the operational efficiency and PKI consolidation.
But if you are only running a couple of windows machines and have little complexity, I would not argue against ADCS being the best choice, it is. It all depends on how large or complex your use case.
